### PR TITLE
feat: refactor SG parser to use event classes

### PR DESCRIPTION
**Pull request by yujia21**
## Issue

[6011](https://github.com/electricitymaps/electricitymaps-contrib/issues/6011)

## Description

This PR updates the SG parser to use event classes as per new parser guidelines. 

Note to reviewer: prior to the change, the SG parser returned a dictionary, but with the change it will return a list of one dictionary. Not sure if this is the ideal pattern for parsers returning just one Event, happy to do a change.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: parsers/SG.py
Comment (VIKTORVAV99) [inline parsers/SG.py:209]: What is going on here? We kind of need solar data...

File: parsers/SG.py
Comment (VIKTORVAV99) [inline parsers/SG.py:213]: While functional the purpose of the Parser classes is to unify and remove unneeded code from the parsers. 

For example this piece of code on line 197:
```python
        if mapped_type:
            generation_by_type[TYPE_MAPPINGS[gen_type]] += gen_mw
```

Should be:
```python
        if mapped_type:
            mix.add_value([TYPE_MAPPINGS[gen_type]], gen_mw)
```
Where mix is the production mix.
This is to ensure we reduce the floating point errors we might introduce from doing math operations and to ensure that if one of the values are negative it's corrected before it's summed with the other values.

File: parsers/SG.py
Comment (VIKTORVAV99) [inline parsers/SG.py:190]: We should get rid of this default dict.

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
Looks like a good start!

Some things need to be changed though before we can merge this.

**Comment by q--** [comment]
This pull request reminded me again that I needed to bring up that Singapore generation mapping for anything other than solar makes wrong assumptions, and that 'unknown' isn't actually renewable.

https://github.com/electricitymaps/electricitymaps-contrib/issues/6508

Apologies for not filing an issue sooner.

**Comment by yujia21** [comment]
Updated with suggested fixes! 

Btw not sure why but the get_solar function is still throwing an error on my local:
```
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7f0d2db0ffb0>
```
but I see the data seems live in production.

**Comment by VIKTORVAV99** [comment]
Code wise this looks good now, will take a look at the issue linked by q-- before approving thought to see if there is something we need to change in this PR.

**Comment by VIKTORVAV99** [comment]
I'm going to merge this and then we can adress the issue in a follow up PR instead.

**Review by VIKTORVAV99 — APPROVED** [review]

**Pull request merged**
